cypress/integration/seededFlows/articleFlows/commentOnArticle.spec.js

```diff
@@ -631,4 +631,58 @@ describe('Comment on articles', () => {
     cy.findByRole('button', { name: /^Toggle dropdown menu$/i }).click();
     cy.findByRole('link', { name: /^Edit this comment$/i }).should('not.exist');
   });
+
+  it('should toggle aria-pressed button', () => {
+    cy.visitAndWaitForUserSideEffects('/admin_mcadmin/test-article-slug');
+
+    // SVG should exists with Like comment: as title
+    cy.get(`div.inner-comment.comment__details`)
+      .first()
```

Comment:
Tests (blocking): Instead of using classes which can change, prefer finding the comments' section by label text and prefer `findByRole` and leverage aliases.

```javascript
      cy.findByRole('button', { name: /^heart$/i })
        .first()
        .as('likeButton')
```

To do this, modify `app/views/articles/_full_comment_area.html.erb`

```diff
- <section id="comments" data-follow-button-container="true" data-updated-at="<%= Time.current %>" class="text-padding mb-4 border-t-1 border-0 border-solid border-base-10">
+ <section id="comments" aria-labelledby="discussionHeader" data-follow-button-container="true" data-updated-at="<%= Time.current %>" class="text-padding mb-4 border-t-1 border-0 border-solid border-base-10">
    <% if @article.show_comments %>
      <header class="relative flex justify-between items-center mb-6">
-       <h2 class="crayons-subtitle-1">Discussion <span class="js-comments-count" data-comments-count="<%= @article.comments_count %>">(<%= @article.comments_count %>)</span></h2>
+       <h2 id="discussionHeader" class="crayons-subtitle-1">Discussion <span class="js-comments-count" data-comments-count="<%= @article.comments_count %>">(<%= @article.comments_count %>)</span></h2>
```

and in your test

```suggestion
    cy.findByLabelText('Discussion (1)').within(() => {
```

The way I see the test is more like this.

```javascript
  it('should toggle aria-pressed button', () => {
    cy.visitAndWaitForUserSideEffects('/admin_mcadmin/test-article-slug');

    // SVG should exists with Like comment: as title
    cy.findByRole('img', { name: 'Like comment:' });

    cy.findByLabelText('Discussion (1)').within(() => {
      cy.findByRole('button', { name: /^heart$/i })
        .first()
        .as('likeButton')
        .and('have.attr', 'aria-pressed', 'false')
        .and('not.have.class', 'reacted');

      cy.get('@likeButton').within(() => {
        cy.get('span.reactions-count').should('have.text', '0');
        cy.get('span.reactions-label').should(($span) => {
          expect($span.text().trim()).equal('Like');
        });
      });

      // React on comment
      cy.get('@likeButton').click();

      // Text should change and react class should be added along with aria-pressed
      cy.get('@likeButton')
        .and('have.attr', 'aria-pressed', 'true')
        .and('have.class', 'reacted');
      cy.get('@likeButton').within(() => {
        cy.get('span.reactions-count').should('have.text', '1');
        cy.get('span.reactions-label').should(($span) => {
          expect($span.text().trim()).equal('like');
        });
      });

      // Unreact on comment
      cy.get('@likeButton').click();

      // Text should change and react class should be added along with aria-pressed
      cy.get('@likeButton')
        .and('have.attr', 'aria-pressed', 'false')
        .and('not.have.class', 'reacted');
      cy.get('@likeButton').within(() => {
        cy.get('span.reactions-count').should('have.text', '0');
        cy.get('span.reactions-label').should(($span) => {
          expect($span.text().trim()).equal('Like');
        });
      });
    });
  });
```

Reply:
@nickytonline Thanks for the suggestion, will make these changes 😄

Reply:
@nickytonline `findByLabelText` is not working with `aria-labelledby` 🙁

Reply:
Any other alternative to this?

Reply:
I think this would work:

```js
cy.findByRole('region', { name: 'Discussion (1)' })
```

BUT...

Labelling this section means we are making a change to the landmarks exposed to screen reader users. This section will now be interpreted as a ["region" landmark](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Region_role), which by necessity needs to make sense when separated from the main content of the page. I'm not sure the comments section qualifies as a region for that reason, as it's intrinsically linked to the article content.

Instead, I'd suggest using the existing data-testid on the inner container, and querying it with

```js
cy.findByTestId('comments-container').within...
```

I think this will get you the same scoping of the page queries that you're looking for, without adding anything additional to that comment container HTML. Let me know how you get on though!

Reply:
@aitchiss Thanks! This worked!!